core/src/main/java/com/linecorp/armeria/client/HttpResponseDecoder.java

```diff
@@ -221,12 +222,14 @@ public boolean tryWrite(HttpObject o) {
                     logBuilder.startResponse();
 
                     assert o instanceof HttpHeaders;
-                    final HttpHeaders headers = (HttpHeaders) o;
-                    final HttpStatus status = headers.status();
 
-                    if (status != null && status.codeClass() != HttpStatusClass.INFORMATIONAL) {
-                        state = State.WAIT_DATA_OR_TRAILERS;
-                        logBuilder.responseHeaders(headers);
+                    if (o instanceof ResponseHeaders) {
```

Comment:
Should we throw an exception if it's `RequestHeaders` here? I guess it's an armeria programming bug?

Reply:
Yeah, now throwing `IllegalStateException`.

Reply:
Modified again to just use assertions.